crates/shared/src/price_estimation.rs

```diff
@@ -138,6 +138,85 @@ impl FromStr for PriceEstimator {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct NativePriceEstimators(Vec<NativePriceEstimator>);
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub enum NativePriceEstimator {
+    GenericPriceEstimator(PriceEstimatorKind),
+    OneInchSpotPriceApi,
+}
+
+impl NativePriceEstimators {
+    fn none() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn as_slice(&self) -> &[NativePriceEstimator] {
+        &self.0
+    }
+}
+
+impl Default for NativePriceEstimators {
+    fn default() -> Self {
+        Self(vec![NativePriceEstimator::GenericPriceEstimator(
+            PriceEstimatorKind::Baseline,
+        )])
+    }
+}
+
+impl Display for NativePriceEstimators {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let mut it = self.as_slice().iter();
+        if let Some(kind) = it.next() {
+            write!(f, "{kind:?}")?;
+            for kind in it {
+                write!(f, ",{kind:?}")?;
+            }
+            Ok(())
+        } else {
+            f.write_str("None")
+        }
```

Comment:
Itertools has your back here:

```suggestion
        if self.as_slice().is_empty() {
            return f.write_str("None");
        }
        
        write!(f, "{:?}", self.as_slice().iter().format(","))
```

Reply:
Fixed in #1885